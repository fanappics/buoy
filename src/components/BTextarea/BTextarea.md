### Example usage

```js
let text = 'My default message'

<b-textarea id="bta1" label="My Textarea" v-model="text" placeholder="You deleted my default message... :(" />
```

### Resizable

```js
<b-textarea id="bta2" label="MyTextarea" placeholder="Type something..." resizable />
```

### Disabled

```js
let text = 'My default message'

<b-textarea id="bta3" label="My Textarea" disabled />
```

### Rows

```js
let text = 'My default message...\n\ncontinuing on line 3.'

<b-textarea id="bta4" label="My Textarea" v-model="text" rows="3" />
```

### Validation

```js
<b-textarea id="bta5" label="My Textarea" placeholder="Type something to continue..." required />
```

### Validation with BValidator

```js
  let textAreasValid = false;

  <b-validator scope="textarea-validator" v-model="textAreasValid">
    <b-textarea id="bta6" label="Textarea 1" validation-name="textarea1" max-length="5" />
    <br />
    <b-textarea id="bta7" label="Textarea 2" validation-name="textarea2" min-length="5" required />
    <br />
    <b-button :disabled="!textAreasValid">Submit</b-button>
  </b-validator>
```
