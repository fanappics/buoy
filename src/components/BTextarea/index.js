// import BTextInput from './BTextInput.vue'
import BTextarea from './BTextarea.vue'

/* istanbul ignore next */
BTextarea.install = function install (Vue) {
  Vue.component(BTextarea.name, BTextarea)
}

export default BTextarea
