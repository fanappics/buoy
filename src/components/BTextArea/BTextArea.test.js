import { shallow } from 'vue-test-utils'
import BTextarea from './BTextarea'

describe('BTextarea', () => {
  it('should render component', () => {
    const wrapper = shallow(BTextarea)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with a label', () => {
    const wrapper = shallow(BTextarea, {
      propsData: {
        label: 'test'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with label and asterisk', () => {
    const wrapper = shallow(BTextarea, {
      propsData: {
        label: 'test',
        required: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have an aria-label if label is hidden', () => {
    const wrapper = shallow(BTextarea, {
      propsData: {
        label: 'test',
        hideLabel: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
