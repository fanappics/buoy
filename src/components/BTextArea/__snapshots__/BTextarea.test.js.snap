// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BTextarea should have an aria-label if label is hidden 1`] = `
<div class="b-input-group">
    <!---->
    <textarea aria-label="test" cols="30" rows="5"></textarea>
    <div class="b-input-group--error"><span></span></div>
</div>
`;

exports[`BTextarea should render component 1`] = `
<div class="b-input-group">
    <label><strong> 
      <!----></strong></label>
    <textarea cols="30" rows="5"></textarea>
    <div class="b-input-group--error"><span></span></div>
</div>
`;

exports[`BTextarea should render component with a label 1`] = `
<div class="b-input-group">
    <label><strong> 
      test<!----></strong></label>
    <textarea cols="30" rows="5"></textarea>
    <div class="b-input-group--error"><span></span></div>
</div>
`;

exports[`BTextarea should render component with label and asterisk 1`] = `
<div class="b-input-group">
    <label><strong> 
      test<span aria-label="Required">*</span></strong></label>
    <textarea cols="30" rows="5" required="required"></textarea>
    <div class="b-input-group--error"><span></span></div>
</div>
`;
